learning-progression-with-la-schema.md

```diff
@@ -1058,6 +1058,10 @@ Program state. List with examples.
 ##### Clock cycles
 [[toc](#table-of-contents)]
 
+Synchronous circuits are circuits in which a clock governs when a circuit will change to its next state. This coordination is necessary in complex circuits like CPUs, because some operations may take longer to complete than others. This can create a bug known as a [race condition](https://en.wikipedia.org/wiki/Race_condition), in which erroneous output is produced because the timing of the circuit's change to the next state was not consistent throughout the circuit.
```

Comment:
@iconoptic, _race condition_ refers to the uncertainty of the result of an operation in the context of multi-threaded execution. Because threads can be interleaved/preempted on the processor-instruction level (i.e. at a higher granularity than program code, meaning that one line of program can compile into several assembly instructions), when multiple threads execute the same _unprotected_ (with a _semaphore_ or other _mutex_ device) section of code, the results are non-deterministic.

Reply:
@iconoptic, on the hardware level, something like the _race condition_ happens in _multi-stable_ circuits. For example, the [flip-flop](https://en.wikipedia.org/wiki/Flip-flop_(electronics)) is a [_bi-stable_](https://en.wikipedia.org/wiki/Multivibrator#Bistable) circuit.